text/XXXX-config_revision.md

```diff
@@ -0,0 +1,268 @@
+
+```yaml
+API:
+  allowedorigins:
+  - electron://local.qri.io
+  - http://localhost:2505
+  enabled: true
+  address: "/ip4/0.0.0.0/tcp/2503"
+  readonly: false
+  serveremotetraffic: false
+CLI:
+  colorizeoutput: true
+  pager: ""
+Logging:
+  levels:
+    lib: info
+P2P:
+  address: "/ip4/0.0.0.0/tcp/2501"
+  addrs: null
+  enabled: true
+  peerid: QmdJQpcmGMkNtKYv5bviZ7z9kPh7uLKSMzqdSbemYs3XpH
+  privkey: <base64-encoded private key>
+  qribootstrapaddrs:
+  - /ip4/35.193.162.149/tcp/4001/ipfs/QmTZxETL4YCCzB1yFx4GT1te68henVHD1XPQMkHZ1N22mm
+  - /ip4/35.239.80.82/tcp/4001/ipfs/QmdpGkbqDYRPCcwLYnEm8oYGz2G9aUZn9WwPjqvqw3XUAc
+  - /ip4/35.225.152.38/tcp/4001/ipfs/QmTRqTLbKndFC2rp6VzpyApxHCLrFV35setF1DQZaRWPVf
+  - /ip4/35.202.155.225/tcp/4001/ipfs/QmegNYmwHUQFc3v3eemsYUVf3WiSg4RcMrh3hovA5LncJ2
+  - /ip4/35.238.10.180/tcp/4001/ipfs/QmessbA6uGLJ7HTwbUJ2niE49WbdPfzi27tdYXdAaGRB4G
+  - /ip4/35.238.105.35/tcp/4001/ipfs/Qmc353gHY5Wx5iHKHPYj3QDqHP4hVA1MpoSsT6hwSyVx3r
+  - /ip4/35.239.138.186/tcp/4001/ipfs/QmT9YHJF2YkysLqWhhiVTL5526VFtavic3bVueF9rCsjVi
+  - /ip4/35.226.44.58/tcp/4001/ipfs/QmQS2ryqZrjJtPKDy9VTkdPwdUSpTi1TdpGUaqAVwfxcNh
+Profile:
+  color: ""
+  created: "2020-02-12T04:29:18Z"
+  description: ""
+  email: brendan+test@qri.io
+  homeurl: ""
+  id: QmSyDX5LYTiwQi861F5NAwdHrrnd1iRGsoEvCyzQMUyZ4W
+  name: ""
+  peername: b5
+  photo: https://qri-user-images.storage.googleapis.com/1570029824064.png
+  poster: /ipfs/QmSN2yHp4qFy1xLCRVQNJKAjgbwHYPAho9fT7C3vAM426c
+  privkey: <base64-encoded private key>
+  thumb: ""
+  twitter: ""
+  type: peer
+  updated: "2018-04-19T18:10:46.627471799-04:00"
+RPC:
+  enabled: true
+  address: "/ip4/0.0.0.0/tcp/2504"
+Registry:
+  location: https://registry.qri.cloud
+Remote:
+  acceptsizemax: 0
+  accepttimeoutms: 0
+  allowremoves: false
+  enabled: false
+  requireallblocks: false
+Remotes: null
+Repo:
+  type: fs
+Revision: 2
+Filesystems:
+- type: ipfs
+  options:
+    path: ./ipfs
+    api: true
+    pubsub: false
+- type: local
+- type: http
+```
+
+### Expanding Store to Filesystems
+
+Currently we hard-code qri filesystem (`qfs`) configuration details. qfs spins up a multiplexed-filesystem by default, composing multiple file-like persistence layers using prefix demuxing. If we expand "Store" to "Filesystem", we can make this fully configurable. The default configuration qri is using would look like this:
+
+```yaml
+Filesystems:
+- type: ipfs
+  config:
+    path: ~/.ipfs
+    api: true
+    pubsub: false
+- type: local
+- type: http
+```
+
+The first win: users could remove `local` or `http` to disable qri's access to the local filesystem or HTTP for the purpose of reading data. We can also expand for later filesystems & options. an `s3` implementation would be nice. this is also how we'd do `dat` integration.
+
+We could use a convention of the default write destination being the first listed filesystem, and present a big warning if users configure anything other than `ipfs` in the first position.
```

Comment:
Could be better to explicitly having a `write: true` flag in the filesystem's config and warn if no flag is specified or if it's not in specified under `ipfs`. Then it's easier for a user to know what's going on just by looking at the config.

Reply:
Agree, but I think we should save it for later. If multiple filestores set `write: true`, how do we know which destination to save it to? For now I think going with the convention of first-one-saves lets us delay dealing with this until later, and won't require deprecating a configuration field

Reply:
We can also just say for now: "qri only saves to IPFS". Then order wouldn't matter, and we can show an error if an IPFS filestore is missing